allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -231,7 +234,10 @@ def forward(self,  # type: ignore
         question_table_similarity_max_score, _ = torch.max(question_table_similarity, 2)
         # (batch_size, num_entities, num_question_tokens, num_features)
         linking_features = table['linking']
-        feature_scores = self._linking_params(linking_features).squeeze(3)
+        if self._linking_params is not None:
+            feature_scores = self._linking_params(linking_features).squeeze(3)
```

Comment:
I'd probably just do `linking_scores += feature_scores` after this line, instead of creating a tensor of zeros that never gets used.  You can have a check below, where `feature_scores` gets added to the output dict, for `self._linking_params is not None`, like you have here.

Reply:
done.